src/nccl_ofi_rdma.cpp

```diff
@@ -2892,16 +2892,16 @@ static inline int reg_mr_on_device(nccl_net_ofi_rdma_domain_t *domain,
 	}
 
         if (nccl_ofi_idpool_active(key_pool)) {
-		ret_handle->mr_key =nccl_ofi_idpool_allocate_id(key_pool);
-		if (OFI_UNLIKELY(ret_handle->mr_key < 0)) {
+		int key = nccl_ofi_idpool_allocate_id(key_pool);
```

Comment:
Let's assume we're going to return a larger type here eventually from `nccl_ofi_idpool_allocate_id` and use `auto` here so we don't miss this later.